### PR TITLE
[IE CLDNN] Implement nGraph function support in QueryNetwork()

### PR DESCRIPTION
**Pull request by mletavin**
Align GPU plugin QueryNetwork() functionality in terms of nGraph support with CPU plugin

### CODE REVIEW

File: inference-engine/src/cldnn_engine/cldnn_engine.cpp [hunk 1]
```diff
@@ -270,72 +271,207 @@ void clDNNEngine::SetConfig(const std::map<std::string, std::string> &config) {
...
+
+        std::unordered_set<std::string> splitNames;
+        std::unordered_set<std::string> concatNames;
+        std::unordered_set<std::string> depLayerNames;
+
+        std::vector<std::shared_ptr<ngraph::Node>> splits;
+        std::vector<std::shared_ptr<ngraph::Node>> concats;
+        std::vector<std::shared_ptr<ngraph::Node>> nextLayerDependent;
+        
+        for (CNNNetworkIterator itLayer{clonedNetwork.get()};
```
Comment (ilya-lavrenov) [inline inference-engine/src/cldnn_engine/cldnn_engine.cpp:300]: does it really work? No conversion from ngraph to old representation is applied

**Comment by mikhail-treskin** [comment]
From tests perspective everything is OK

File: inference-engine/src/cldnn_engine/cldnn_engine.cpp (hunk 1 above)
Comment (mletavin) [reply inference-engine/src/cldnn_engine/cldnn_engine.cpp:300]: Sorry, I didn't catch what you mean exactly.
If you mean CNNLayer objects creation, than it was done by executing convertFunctionToICNNNetwork() inside CloneNetwork() method, which was called above.
The code definitely works - added tests pass and any v10 IR I've tested so far passed successfully.

Comment (ilya-lavrenov) [reply inference-engine/src/cldnn_engine/cldnn_engine.cpp:300]: > executing convertFunctionToICNNNetwork() inside CloneNetwork() method

This is not right, `CloneNetwork` does not perform conversion. Implicit conversion is removed from IE (except `setBatchSize`). See https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/legacy_api/src/ie_util_internal.cpp#L153-L161, not conversion is applied

Comment (mletavin) [reply inference-engine/src/cldnn_engine/cldnn_engine.cpp:304]: Done.

File: inference-engine/src/cldnn_engine/cldnn_engine.cpp [hunk 2]
```diff
@@ -274,72 +275,207 @@ void clDNNEngine::SetConfig(const std::map<std::string, std::string> &config) {
...
-        bool supported = true;
-        for (DataPtr out : (*cnl)->outData) {
-            for (auto ol : getInputTo(out)) {
-                if (res.supportedLayersMap.find(ol.second->name) == res.supportedLayersMap.end()) {
+        for (auto&& layerName : supported) {
+            res.supportedLayersMap.emplace(layerName, GetName());
+        }
+    } else {
+        std::vector <CNNLayer::Ptr> concats;
+        std::vector <CNNLayer::Ptr> nextLayerDependent;
```
Comment (vladimir-paramuzov) [inline inference-engine/src/cldnn_engine/cldnn_engine.cpp:423]: nit: code style `std::vector<CNNLayer::Ptr>`

File: inference-engine/src/cldnn_engine/cldnn_engine.cpp [hunk 3]
```diff
@@ -274,72 +275,207 @@ void clDNNEngine::SetConfig(const std::map<std::string, std::string> &config) {
...
+            cnl++) {
+            bool supported = true;
+            for (DataPtr out : (*cnl)->outData) {
+                for (auto ol : getInputTo(out)) {
+                    if (res.supportedLayersMap.find(ol.second->name) == res.supportedLayersMap.end()) {
+                        supported = false;
+                    }
+                }
+            }
+            std::cout << (*cnl)->name << " is " << (supported ? "GPU" : "CPU") << std::endl;
```
Comment (vladimir-paramuzov) [inline inference-engine/src/cldnn_engine/cldnn_engine.cpp:474]: Do we still need this printout?

File: inference-engine/src/cldnn_engine/cldnn_engine.cpp (hunk 2 above)
Comment (mletavin) [reply inference-engine/src/cldnn_engine/cldnn_engine.cpp]: Fixed

File: inference-engine/src/cldnn_engine/cldnn_engine.cpp (hunk 3 above)
Comment (mletavin) [reply inference-engine/src/cldnn_engine/cldnn_engine.cpp]: Removed

**Comment by ilya-lavrenov** [comment]
@apankratovantonp has also implemented `synthetic.cpp` tests. Should not we enable them as well?

**Pull request merged**